proteuslib/unit_models/reverse_osmosis_0D.py

```diff
@@ -622,16 +627,22 @@ def eq_N_Re_io(b, t, x):
                 elif x == 'out':
                     prop_io = b.feed_side.properties_out[t]
                 return (b.N_Re_io[t, x] ==
-                        sum(prop_io.flow_mass_phase_comp['Liq', j] for j in b.solute_list)
+                        sum(prop_io.flow_mass_phase_comp['Liq', j] for j in b.config.property_package.component_list)
                         / b.area_cross
                         * b.dh
                         / prop_io.visc_d_phase['Liq'])
 
-            @self.Constraint(doc="Hydraulic diameter")  # TODO: add detail related to spacer geometry
+            @self.Constraint(doc="Hydraulic diameter")  # eqn. 17 in Schock & Miquel, 1987
             def eq_dh(b):
                 return (b.dh ==
-                        2 * (b.channel_height * b.width)
-                        / (b.channel_height + b.width))
+                        4 * b.spacer_porosity
+                        / (2 / b.channel_height
+                           + (1 - b.spacer_porosity) * 4 / b.df))
+
+            # Filament diameter cannot be larger than channel height
+            @self.Constraint(doc="filament diameter constraint")
+            def eq_df(b):
+                return b.df <= b.channel_height
```

Comment:
This constraint should be removed. While it should be always satisfied. we typically don't add constraints like this. For example we don't have a constraint that the outlet flux should be less than inlet flux and you can imagine many additional constraints we can right that should always be true. 

This case is slightly different than the example since these are both supposed to specified by the user, but if the user fixed them to wrong values they would initially just get that the model is infeasible (albeit they could dig into it and figure out it is this constraint). However, in general, we don't provide checks and/or constraints on whether user provided parameters are feasible, that responsibility is on the user. 

Also thinking about this more, typically we don't optimize the spacer geometry. Usually that would require another CFD study, for example df = 1/2 * h is a condition for using Hoek's relationship, therefore we should include the calculation of df as a constraint rather than a user provided input. I'm unsure about the other relationships.

Reply:
Setting a constraint that includes the calculation of df would just be something like ` df == df_h_ratio * h'. In this case we'd build another variable (`df_h_ratio`) with bounds between some epsilon value and 1. In my view, that would be unnecessary since the current constraint essentially does the same thing without needing to set any ratio. I don't think that this case is comparable to the example of setting a constraint for outlet flux < inlet flux because outlet flux should never be greater than inlet flux, unless something was computed incorrectly in the model. 

However, the filament diameter is directly related to channel height, and in imagining some case where the user were not to fix _both_ the height and filament diameter simultaneously, I originally thought we might need this constraint. Although we aren't considering optimizing spacer geometry in the detailed way that CFD studies would, I still think we can leave this open for the user in case they'd want the model to yield the df/h ratio (for example). 

Overall, I have no issues with removing this constraint if we are absolutely certain that `df` will always be fixed by the user in every scenario; alternatively, would the user set a lower bound and upper bound on df instead?  On the other hand, if we were to keep this constraint, we should also have more equations for the Sherwood number and friction factor, which would be related to spacer dimensions (which would be another PR).